.github/workflows/conda-forge-install-and-test.yaml

```diff
@@ -0,0 +1,42 @@
+name: Install all deps from conda-forge and run pytest
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release/*'
+    tags:
+      - 'v*'
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    env:
+      DISPLAY: ':99.0'
+      OS: ${{ matrix.os }}
+
+    steps:
+    - name: checkout source code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
+    - name: install dependencies
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        activate-environment: qcodesforge
+        environment-file: .github/static/environment_forge_full.yml
+        auto-activate-base: false
+    - name: install qcodes
+      shell: bash -l {0}
+      run: |
+        conda info
+        pip install . --no-deps
+    - name: run tests
+      shell: bash -l {0}
+      run: |
+        pytest -n 1
```

Comment:
i would try with `pytest -n 0 --dist no` which runs all tests in the main process. `-n 1` spawns exactly one process that runs all tests